pkg/server/file.go

```diff
@@ -114,22 +114,24 @@ func (t *TranslateServiceServer) UploadTranslationFile(
 		return nil, status.Errorf(codes.InvalidArgument, err.Error())
 	}
 
-	empty, err, done := t.SaveMsgsAndHandleErrors(ctx, params, messages)
-	if done {
-		return empty, err
+	_, err = t.repo.LoadService(ctx, params.serviceID)
+	switch {
+	case errors.Is(err, repo.ErrNotFound):
+		return nil, status.Errorf(codes.NotFound, "service not found")
+	case err != nil:
+		return nil, status.Errorf(codes.Internal, "")
```

Comment:
expensive as redundant?, as `repo.SaveMessages` returns error if no `serviceID` is found, and `repo.LoadMessages` returns empty slice

Reply:
Yes, correct. Will fix it, thanks ;)

Reply:
My bad. We need to focus on the happy path more as it delivers much higher value than chasing insignificant issues like "not found" - Pareto principle.